tests/graphics/canvas_test.py

```diff
@@ -0,0 +1,217 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable
+
+import pytest
+
+from plopp import Node, backends
+from plopp.data import data1d, data2d, data_array, scatter
+from plopp.graphics import imagefigure, linefigure, scatter3dfigure, scatterfigure
+
+
+@dataclass
+class FigureAndData:
+    figure: Callable
+    data: Callable
+
+
+def backend_linefigure(*args, backend=None, **kwargs):
+    backends['2d'] = backend
+    return linefigure(*args, **kwargs)
+
+
+CASES = [
+    FigureAndData(partial(backend_linefigure, backend='matplotlib'), data1d),
+    FigureAndData(partial(backend_linefigure, backend='plotly'), data1d),
+    FigureAndData(partial(imagefigure, cbar=True), data2d),
+    FigureAndData(partial(scatterfigure, x='x', y='y', cbar=True), scatter),
+    FigureAndData(partial(scatter3dfigure, x='x', y='y', z='z', cbar=True), scatter),
```

Comment:
This looks suspicious. `backend_linefigure` sets the backend. But the other functions don't. Why is that. (I know that there is a fixture that resets the backend.)

Reply:
Yes, it is relying on the fixture that resets the backends.

I basically wanted to run the same tests with all possible cases, including plotly.
Didn't really know how else to do it. I thought about trying to use a fixture, as done [here](https://github.com/scipp/plopp/blob/main/tests/backends/plotly/conftest.py#L9), but wasn't sure how to.

Do you have a suggestion?

Reply:
I think it is ok for tests to select a backend. I was mostly surprised that there is only one function that does it while the other functions just use the global default backend.